xarray/tests/test_dataset.py

```diff
@@ -3495,6 +3495,26 @@ def test_setitem_str_dtype(self, dtype):
 
         assert np.issubdtype(ds.x.dtype, dtype)
 
+    def test_setitem_using_list(self):
+
+        # assign a list of variables
+        var1 = Variable(["dim1"], np.random.randn(8))
+        var2 = Variable(["dim1"], np.random.randn(8))
+        data1 = create_test_data()
+        data2 = data1.copy()
+        data1["A"] = var1
+        data1["B"] = var2
+        data2[["A", "B"]] = [var1, var2]
+        assert_identical(data1, data2)
+        # assign a list of dataset arrays
+        dv = 2 * data2[["A", "B"]]
+        data1[["C", "D"]] = [d.variable for d in dv.data_vars.values()]
+        data2[["C", "D"]] = dv
+        assert_identical(data1, data2)
```

Comment:
```suggestion
        # assign a list of numpy arrays
        actual[["C", "D"]] = [var1.values, var2.values]
        expected[["C", "D"]] = expected[["A", "B"]]
        assert_identical(actual, expected)
```

Reply:
Good point — it doesn't align on dim. I'm getting:

```python
AssertionError: Left and right Dataset objects are not identical
Differing dimensions:
    (C: 8, D: 8, dim1: 8, dim2: 9, dim3: 10, time: 20) != (dim1: 8, dim2: 9, dim3: 10, time: 20)
Coordinates only on the left object:
  * C        (C) float64 0.2328 0.9857 0.9737 0.4723 0.3088 -1.016 0.1256 -0.867
  * D        (D) float64 -0.378 -0.152 -0.9274 -0.6869 ... 0.7534 2.461 0.4231
Data variables only on the right object:
    C        (dim1) float64 0.2328 0.9857 0.9737 0.4723 ... -1.016 0.1256 -0.867
    D        (dim1) float64 -0.378 -0.152 -0.9274 ... 0.7534 2.461 0.4231
```

I don't think we need the numpy array test unless we're expecting it to align on dims. Are we?

Reply:
(Though in the current code `test2` doesn't exist, so we need to change that to merge now)